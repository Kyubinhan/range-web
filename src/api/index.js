// import { v4 } from 'node-uuid';

import paginatedFakeAgreements from './fakeAgreements';

const fakeDatabase = {
  paginatedAgreements: paginatedFakeAgreements,
};

const delay = ms => new Promise(resolve => setTimeout(resolve, ms));

export const fetchAgreements = filter =>
  delay(500).then(() => {
    switch (filter) {
      case 'all':
        return fakeDatabase.paginatedAgreements;
      // case 'complete':
      //   return fakeDatabase.paginatedAgreements.agreements.filter(t => !t.draft);
      default:
        throw new Error(`Unknown filter: ${filter}`);
    }
  });

export const s = {};

// export const addTodo = (text) =>
//   delay(500).then(() => {
//     const todo = {
//       id: v4(),
//       text,
//       completed: false,
//     };
//     fakeDatabase.todos.push(todo);
//     return todo;
//   });

// export const toggleTodo = (id) =>
//   delay(500).then(() => {
//     const todo = fakeDatabase.todos.find(t => t.id === id);
//     todo.completed = !todo.completed;
//     return todo;
//   });
