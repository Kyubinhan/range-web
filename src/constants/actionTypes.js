export const STORE_SSO_AUTH_DATA = 'STORE_SSO_AUTH_DATA';
export const STORE_USER = 'STORE_USER';
export const SIGN_OUT = 'SIGN_OUT';

export const REQUEST = 'REQUEST';
export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';
export const SUCCESS_PAGINATED = 'SUCCESS_PAGINATED';

export const STORE_AGREEMENTS = 'STORE_AGREEMENTS';
export const SET_AGREEMENT_FILTER = 'SET_AGREEMENT_FILTER';

export const STORE_PLAN = 'STORE_PLAN';
// export const STORE_ALBUM = 'STORE_ALBUM';
// export const STORE_ARTIST = 'STORE_ARTIST';
// export const STORE_LIBRARY_ALBUMS = 'STORE_LIBRARY_ALBUMS';
// export const FAVORITE_ALBUM = 'FAVORITE_ALBUM';
// export const UNFAVORITE_ALBUM = 'UNFAVORITE_ALBUM';

// export const SET_FILTER = 'SET_FILTER';
// export const SELECT_ARTIST = 'SELECT_ARTIST';
