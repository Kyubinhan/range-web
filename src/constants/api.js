export const SSO_BASE_URL = "https://dev-sso.pathfinder.gov.bc.ca/";
export const SSO_REALM_NAME = "mobile";
export const SSO_BASE_AUTH_ENDPOINT = `${SSO_BASE_URL}auth/realms/mobile/protocol/openid-connect/auth`;
export const SSO_CLIENT_ID = "range-test";
export const SSO_REDIRECT_URI = `${window.location.origin}/login`;

export const SSO_AUTH_ENDPOINT = SSO_BASE_AUTH_ENDPOINT
  + "?response_type=code"
  + `&client_id=${SSO_CLIENT_ID}`
  + `&redirect_uri=${SSO_REDIRECT_URI}`;

export const GET_TOKEN = `auth/realms/${SSO_REALM_NAME}/protocol/openid-connect/token`;
export const REFRESH_TOKEN = `auth/realms/${SSO_REALM_NAME}/protocol/openid-connect/token`

export const BASE_URL = 'http://api-range-myra-dev.pathfinder.gov.bc.ca/v1';
export const AGREEMENT = '/agreement';
export const REFERENCES = '/reference';
export const CLIENT = '/client';
export const STATUS = '/status';
export const ZONE = '/zone';