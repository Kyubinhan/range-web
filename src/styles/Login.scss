.login {
  font-family: Lato;

  &__no-support-browser {
    background: $accent-color;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    padding: 15px 0;

    &__title {
      font-size: 1.5rem;
      font-weight: 500;
      margin-bottom: 5px;
    }
  }
  &__header {
    display: flex;
    justify-content: center;
    align-items: center;
    background: $primary-color;
    height: 60px;

    &__logo {
      height: 60px;
    }
  }
  &__paragraph-cell {
    max-width: 45%;
    display: flex;
    flex-direction: column;
    justify-content: center;

    @include mobile {
      max-width: 100%;
      margin: 10px 0;
    }
  }
  &__title-spacer {
    width: 200px;
    margin-bottom: 15px;
    border: 2px solid #e7e7e7;
  }
  &__paragraph1 {
    background: url('~/signin_background.jpg') no-repeat;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 50px 0;
  }
  &__paragraph2 {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 50px 0;
    box-shadow: 0 2px 20px 0 rgba(35,64,117,0.1);

    &__title {
      font-size: 2.3rem;
      color: $primary-color;
      line-height: initial;
      margin-bottom: 10px;
    }
    &__text {
      max-width: 600px;
      font-size: 1.2rem;
      line-height: 25px;
      text-align: center;
      padding: 20px;
    }
  }
  &__paragraph3,
  &__paragraph4,
  &__paragraph5 {
    padding: 50px 0;
    box-shadow: 0 2px 20px 0 rgba(35,64,117,0.1);

    &__content {
      display: flex;
      justify-content: space-between;
      flex-wrap: wrap;
    }
    &__title {
      font-size: 2rem;
      color: $primary-color;
      line-height: initial;
      margin-bottom: 10px;
    }
    &__text {
      font-size: 1.2rem;
      line-height: 20px;
    }
    &__image {
      max-width: 100%;
      height: auto;
      display: block;
      border-radius: 5px;
    }
  }
  &__footer {
    display: flex;
    justify-content: flex-end;
    padding: 15px 0px;
  }
  &__divider {
    width: 1px;
    border-right: 1px solid #CCCCCC;
    margin: 0 15px 0 11px;
    padding: 2px;
  }
  &__signin {
    &__container {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      padding: 30px;
      min-width: 400px;
      background: white;
      border-radius: 5px;
      font-family: Lato;
    }
    &__title {
      font-size: 2.3rem;
      line-height: initial;
      color: #234075;
    }
    &__text1 {
      font-size: 1.1rem;
      text-align: center;
      margin-top: 5px;
    }
    &__text2 {
      font-size: 1.1rem;
      text-align: center;
      margin-top: 20px;
    }
    &__text3 {
      font-size: 1.1rem;
    }
    &__error {
      margin-top: 15px;
      width: 100%;
      text-align: center;
    }
    &__link-container {
      align-self: stretch;
      text-align: right;
      margin-top: 25px;
      border-top: 1px solid lightgray;
      padding-top: 15px;

      > div {
        display: inline;
        cursor: pointer;
      }
    }
  }
}
