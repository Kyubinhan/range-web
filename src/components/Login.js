import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button, Segment } from 'semantic-ui-react';
import { Loading } from './common';
import { SSO_LOGIN_ENDPOINT, SSO_IDIR_LOGIN_ENDPOINT, SSO_BCEID_LOGIN_ENDPOINT } from '../constants/API';
import { ELEMENT_ID, IMAGE_SRC } from '../constants/variables';
import { storeAuthData } from '../actions';
import { fetchUser } from '../actionCreators';
import { getIsFetchingUser } from '../reducers/rootReducer';
import { APP_NAME } from '../constants/strings';
import { detectIE } from '../utils';

const propTypes = {
  storeAuthData: PropTypes.func.isRequired,
  fetchUser: PropTypes.func.isRequired,
  isFetchingUser: PropTypes.bool.isRequired,
};

export class Login extends Component {
  // Sets up localstorage listener for cross-tab communication
  // since the authentication requires the user to be redirected
  // to another page and then redirected back to a return URL with the token.
  componentDidMount() {
    window.addEventListener('storage', this.storageEventListener);
  }

  componentWillUnmount() {
    window.removeEventListener('storage', this.storageEventListener);
  }

  storageEventListener = (event) => {
    const { storeAuthData, fetchUser } = this.props;
    const authData = JSON.parse(localStorage.getItem(event.key));

    // store the auth data in Redux store
    storeAuthData(authData);

    fetchUser();
  }

  openNewTab = link => window.open(link, '_black')
  onLoginBtnClick = () => this.openNewTab(SSO_LOGIN_ENDPOINT)
  onIdirLoginBtnClick = () => this.openNewTab(SSO_IDIR_LOGIN_ENDPOINT)
  onBceidLoginBtnClick = () => this.openNewTab(SSO_BCEID_LOGIN_ENDPOINT)

  render() {
    const { isFetchingUser } = this.props;
    const isIE = detectIE();

    return (
      <section className="login">
        {isIE &&
          <article className="login__no-support-browser">
            <div className="login__no-support-browser__title">
              Your internet browser is not supported.
            </div>
            <div>
              <div>
                Please visit {APP_NAME} using a supported browser:
                <a href="https://www.google.com/chrome" target="_blank" rel="noopener noreferrer"> Google Chrome,</a>
                <a href="https://www.mozilla.org/en-US/firefox/new" target="_blank" rel="noopener noreferrer"> Firefox,</a>
                <a href="https://support.apple.com/en-ca/safari" target="_blank" rel="noopener noreferrer"> Safari </a>
                (Mac only).
              </div>
              If you choose to continue with this browser the application may not work as intended.
            </div>
          </article>
        }
        <article className="login__header">
          <img className="login__header__logo" src={IMAGE_SRC.LOGIN_LOGO} alt="Logo" />
          <div className="login__header__title">{APP_NAME}</div>
        </article>
        <article className="login__paragraph1">
          <Segment basic>
            <Loading active={isFetchingUser} />
            <div className="login__signin__container">
              <div className="login__signin__title">Sign In</div>
              <div className="login__signin__text1">to continue to {APP_NAME}</div>
              <div className="login__signin__text2">We use the BCeID for authentication.</div>
              <a
                className="login__signin__text3"
                href="https://portal.nrs.gov.bc.ca/web/client/bceid"
                target="_blank"
                rel="noopener noreferrer"
              >
                Learn more about BCeID here.
              </a>
              <Button
                id={ELEMENT_ID.LOGIN_BCEID_BUTTON}
                className="login__signin__button"
                primary
                fluid
                style={{ height: '50px' }}
                onClick={this.onBceidLoginBtnClick}
              >
                Login as Agreement Holder
              </Button>
              <div className="login__signin__link-container">
                <div
                  role="button"
                  tabIndex="0"
                  onClick={this.onIdirLoginBtnClick}
                >
                  Range Staff Login
                </div>
                <div className="login__line" />
                <div
                  role="button"
                  tabIndex="0"
                  onClick={this.onLoginBtnClick}
                >
                  Admin Login
                </div>
              </div>
            </div>
          </Segment>
        </article>
        <article className="login__paragraph2">
          <div className="login__paragraph2__title">What is {APP_NAME}?</div>
          <div className="login__paragraph2__text">
            We are making it easier for you to submit and amend Range Use Plans. This new service will enable you to submit your plan electronically, save drafts and print versions.
            Your Range Officer is also getting new tools to allow so that together we can manage the land with greater accuracy.
          </div>
        </article>
        <article className="login__paragraph3">
          <div className="container">
            <div className="login__paragraph4__content">
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph3__title">
                  New simplified Range Use Plan across BC.
                </div>
                <div className="login__paragraph3__text">
                  There is now a common Range Use Plan across BC.
                  This common plan will enable greater efficiencies in service delivery.
                </div>
              </div>
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph3__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH3}
                  alt="paragraph3_image"
                />
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph4">
          <div className="container">
            <div className="login__paragraph4__content">
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph4__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH4}
                  alt="paragraph4_image"
                />
              </div>
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph4__title">
                  Submit your Range Use Plan faster then ever.
                </div>
                <div className="login__paragraph4__text">
                  No more mail or paper. When you submit a Range Use Plan for review or amendment your Range Staff Officer can review it and you will know its status.
                </div>
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph5">
          <div className="container">
            <div className="login__paragraph5__content">
              <div className="login__paragraph-cell">
                <hr className="login__title-spacer" />
                <div className="login__paragraph5__title">
                  Easier login with BCeID
                </div>
                <div className="login__paragraph5__text">
                  With your BCeID you can view all of your Range Use Plans and amendments. We have included ways to save time in drafting your work
                </div>
              </div>
              <div className="login__paragraph-cell">
                <img
                  className="login__paragraph5__image"
                  src={IMAGE_SRC.LOGIN_PARAGRAPH5}
                  alt="paragraph5_image"
                />
              </div>
            </div>
          </div>
        </article>
        <article className="login__paragraph6">
          <div className="container">
            <div className="login__footer">
              Copyright
              <div className="login__line" />
              Accessability
              <div className="login__line" />
              Privacy
              <div className="login__line" />
              Disclaimer
            </div>
          </div>
        </article>
      </section>
    );
  }
}

const mapStateToProps = state => (
  {
    isFetchingUser: getIsFetchingUser(state),
  }
);

Login.propTypes = propTypes;
export default connect(mapStateToProps, { storeAuthData, fetchUser })(Login);
