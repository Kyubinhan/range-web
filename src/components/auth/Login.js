import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

import { parseQuery } from '../../handlers';
import { SSO_AUTH_LOGIN_ENDPOINT } from '../../constants/api';
import { LOGIN_LOGO_SRC } from '../../constants/variables';
import { login } from '../../actions/authActions';

const propTypes = {
  login: PropTypes.func.isRequired,
  location: PropTypes.shape({}).isRequired,
  loginState: PropTypes.shape({}).isRequired,
};

export class Login extends Component {
  componentDidMount() {
    const { login, location } = this.props;

    // grab the code from the redirect url
    const { code } = parseQuery(location.search);

    if (code) {
      login(code);
    }
  }

  onLoginBtnClick = () => {
    window.open(SSO_AUTH_LOGIN_ENDPOINT, '_self');
  }

  render() {
    const {
      loginState,
    } = this.props;

    return (
      <div className="login">
        <img
          className="login__image"
          src={LOGIN_LOGO_SRC}
          alt="login-img"
        />

        <div className="login__title">
          My Range Application
        </div>

        <div className="login__button">
          <Button
            id="login-button"
            loading={loginState.isLoading || false}
            disabled={loginState.isLoading || false}
            primary
            fluid
            onClick={this.onLoginBtnClick}
          >
            Login
          </Button>

        </div>
        <a
          className="login__change-link"
          href="https://summer.gov.bc.ca"
          target="_blank"
          rel="noopener noreferrer"
        >
          Is your password expired?
        </a>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    loginState: state.auth,
  }
);

Login.propTypes = propTypes;
export default connect(mapStateToProps, { login })(Login);

