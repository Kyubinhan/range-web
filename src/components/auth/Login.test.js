import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';
import { SSO_AUTH_ENDPOINT } from '../../constants/api';

const props = {};
const setupProps = () => {
  props.login = jest.fn();
  props.loginState = {};
  props.location = {
    search: '',
  };
};

beforeEach(() => {
  setupProps();
});

describe('Login', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('opens a new tab when clicking on Login button', () => {
    global.window.open = jest.fn();
    const wrapper = shallow(<Login {...props} />);

    wrapper.find('Button').simulate('click', {});
    expect(global.window.open).toHaveBeenCalledWith(SSO_AUTH_ENDPOINT, '_self'); 
  });

  describe('Life cycles', () => {
    it('componentDidmount', () => {
      const mockCode = 'hhAvWWCcAztaAfswBN4XHbPPgiplWk8UtY5Lhs';
      props.location = {
        search: `?code=${mockCode}`,
      };
      const wrapper = shallow(<Login {...props} />);
      const { login } = props;
      expect(login).toHaveBeenCalledWith(mockCode);
    });
  });
});