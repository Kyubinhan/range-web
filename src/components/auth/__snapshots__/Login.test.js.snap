// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    login={[Function]}
    loginState={Object {}}
    user={
        Object {
            "id": "id",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="gov-img"
          className="login__image"
          src="https://avatars3.githubusercontent.com/u/916280?s=200&v=4"
/>,
        <div
          className="login__title"
>
          My Range Application
</div>,
        <div
          className="login__header"
>
          Login
</div>,
        <Form
          as="form"
          loading={false}
>
          <div
                    className="login__form"
          >
                    <FormField>
                              <label>
                                        Username
                              </label>
                              <input
                                        id="username"
                                        onChange={[Function]}
                                        placeholder="Enter Username"
                                        type="text"
                                        value=""
                              />
                    </FormField>
          </div>
          <div
                    className="login__form"
          >
                    <FormField>
                              <label>
                                        Password
                              </label>
                              <input
                                        id="password"
                                        onChange={[Function]}
                                        onKeyPress={undefined}
                                        placeholder="Enter Password"
                                        type="password"
                                        value=""
                              />
                    </FormField>
          </div>
          <div
                    className="login__button"
          >
                    <Button
                              as="button"
                              fluid={true}
                              onClick={[Function]}
                              primary={true}
                    >
                              Login
                    </Button>
          </div>
</Form>,
        <a
          className="login__change-link"
          href="https://summer.gov.bc.ca"
          rel="noopener noreferrer"
          target="_blank"
>
          Is your password expired?
</a>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gov-img",
          "className": "login__image",
          "src": "https://avatars3.githubusercontent.com/u/916280?s=200&v=4",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Range Application",
          "className": "login__title",
        },
        "ref": null,
        "rendered": "My Range Application",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "className": "login__header",
        },
        "ref": null,
        "rendered": "Login",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": Array [
            <div
              className="login__form"
>
              <FormField>
                            <label>
                                          Username
                            </label>
                            <input
                                          id="username"
                                          onChange={[Function]}
                                          placeholder="Enter Username"
                                          type="text"
                                          value=""
                            />
              </FormField>
</div>,
            <div
              className="login__form"
>
              <FormField>
                            <label>
                                          Password
                            </label>
                            <input
                                          id="password"
                                          onChange={[Function]}
                                          onKeyPress={undefined}
                                          placeholder="Enter Password"
                                          type="password"
                                          value=""
                            />
              </FormField>
</div>,
            <div
              className="login__button"
>
              <Button
                            as="button"
                            fluid={true}
                            onClick={[Function]}
                            primary={true}
              >
                            Login
              </Button>
</div>,
          ],
          "loading": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField>
                <label>
                                Username
                </label>
                <input
                                id="username"
                                onChange={[Function]}
                                placeholder="Enter Username"
                                type="text"
                                value=""
                />
</FormField>,
              "className": "login__form",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label>
                    Username
</label>,
                  <input
                    id="username"
                    onChange={[Function]}
                    placeholder="Enter Username"
                    type="text"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Username",
                  },
                  "ref": null,
                  "rendered": "Username",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "username",
                    "onChange": [Function],
                    "placeholder": "Enter Username",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormField>
                <label>
                                Password
                </label>
                <input
                                id="password"
                                onChange={[Function]}
                                onKeyPress={undefined}
                                placeholder="Enter Password"
                                type="password"
                                value=""
                />
</FormField>,
              "className": "login__form",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label>
                    Password
</label>,
                  <input
                    id="password"
                    onChange={[Function]}
                    onKeyPress={undefined}
                    placeholder="Enter Password"
                    type="password"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "password",
                    "onChange": [Function],
                    "onKeyPress": undefined,
                    "placeholder": "Enter Password",
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                as="button"
                fluid={true}
                onClick={[Function]}
                primary={true}
>
                Login
</Button>,
              "className": "login__button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": "Login",
                "fluid": true,
                "onClick": [Function],
                "primary": true,
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Is your password expired?",
          "className": "login__change-link",
          "href": "https://summer.gov.bc.ca",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Is your password expired?",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="gov-img"
            className="login__image"
            src="https://avatars3.githubusercontent.com/u/916280?s=200&v=4"
/>,
          <div
            className="login__title"
>
            My Range Application
</div>,
          <div
            className="login__header"
>
            Login
</div>,
          <Form
            as="form"
            loading={false}
>
            <div
                        className="login__form"
            >
                        <FormField>
                                    <label>
                                                Username
                                    </label>
                                    <input
                                                id="username"
                                                onChange={[Function]}
                                                placeholder="Enter Username"
                                                type="text"
                                                value=""
                                    />
                        </FormField>
            </div>
            <div
                        className="login__form"
            >
                        <FormField>
                                    <label>
                                                Password
                                    </label>
                                    <input
                                                id="password"
                                                onChange={[Function]}
                                                onKeyPress={undefined}
                                                placeholder="Enter Password"
                                                type="password"
                                                value=""
                                    />
                        </FormField>
            </div>
            <div
                        className="login__button"
            >
                        <Button
                                    as="button"
                                    fluid={true}
                                    onClick={[Function]}
                                    primary={true}
                        >
                                    Login
                        </Button>
            </div>
</Form>,
          <a
            className="login__change-link"
            href="https://summer.gov.bc.ca"
            rel="noopener noreferrer"
            target="_blank"
>
            Is your password expired?
</a>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gov-img",
            "className": "login__image",
            "src": "https://avatars3.githubusercontent.com/u/916280?s=200&v=4",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Range Application",
            "className": "login__title",
          },
          "ref": null,
          "rendered": "My Range Application",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login__header",
          },
          "ref": null,
          "rendered": "Login",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": Array [
              <div
                className="login__form"
>
                <FormField>
                                <label>
                                                Username
                                </label>
                                <input
                                                id="username"
                                                onChange={[Function]}
                                                placeholder="Enter Username"
                                                type="text"
                                                value=""
                                />
                </FormField>
</div>,
              <div
                className="login__form"
>
                <FormField>
                                <label>
                                                Password
                                </label>
                                <input
                                                id="password"
                                                onChange={[Function]}
                                                onKeyPress={undefined}
                                                placeholder="Enter Password"
                                                type="password"
                                                value=""
                                />
                </FormField>
</div>,
              <div
                className="login__button"
>
                <Button
                                as="button"
                                fluid={true}
                                onClick={[Function]}
                                primary={true}
                >
                                Login
                </Button>
</div>,
            ],
            "loading": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormField>
                  <label>
                                    Username
                  </label>
                  <input
                                    id="username"
                                    onChange={[Function]}
                                    placeholder="Enter Username"
                                    type="text"
                                    value=""
                  />
</FormField>,
                "className": "login__form",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label>
                      Username
</label>,
                    <input
                      id="username"
                      onChange={[Function]}
                      placeholder="Enter Username"
                      type="text"
                      value=""
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Username",
                    },
                    "ref": null,
                    "rendered": "Username",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "username",
                      "onChange": [Function],
                      "placeholder": "Enter Username",
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormField>
                  <label>
                                    Password
                  </label>
                  <input
                                    id="password"
                                    onChange={[Function]}
                                    onKeyPress={undefined}
                                    placeholder="Enter Password"
                                    type="password"
                                    value=""
                  />
</FormField>,
                "className": "login__form",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <label>
                      Password
</label>,
                    <input
                      id="password"
                      onChange={[Function]}
                      onKeyPress={undefined}
                      placeholder="Enter Password"
                      type="password"
                      value=""
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "password",
                      "onChange": [Function],
                      "onKeyPress": undefined,
                      "placeholder": "Enter Password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  as="button"
                  fluid={true}
                  onClick={[Function]}
                  primary={true}
>
                  Login
</Button>,
                "className": "login__button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "Login",
                  "fluid": true,
                  "onClick": [Function],
                  "primary": true,
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Is your password expired?",
            "className": "login__change-link",
            "href": "https://summer.gov.bc.ca",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Is your password expired?",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
